### PR TITLE
Fix stat drop in doubles for single target moves

### LINKED ISSUE #4002: Moves that should drop stats on the user don't in doubles
### Description

mGBA_zsowsJ07Cu

The likely issue: 
https://github.com/cfmnephrite/pokeemerald/blob/095decbd278216893fe9f374a3393c180b5e01c8/src/battle_script_commands.c#L3815

### Version

upcoming

### Upcoming Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #4002

Technically both the single and double battle test check the same thing but since this was a bug I wrote them for both.

### CODE REVIEW

File: test/battle/move_effect/pay_day.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(MoveHasMoveEffect(MOVE_PAY_DAY, MOVE_EFFECT_PAYDAY));
+}
+
+SINGLE_BATTLE_TEST("Single Battle: Pay Day Scatters coins around after it hits")
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/pay_day.c:9]:
```suggestion
SINGLE_BATTLE_TEST("Pay Day Scatters coins around after it hits - singles")
```

File: test/battle/move_effect/pay_day.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    } WHEN {
+        TURN { MOVE(player, MOVE_PAY_DAY); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_PAY_DAY, player);
+        HP_BAR(opponent);
+        MESSAGE("Coins scattered everywhere!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Double Battle: Pay Day Scatters coins around after it hits")
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/pay_day.c:23]:
```suggestion
DOUBLE_BATTLE_TEST("Pay Day Scatters coins around after it hits - doubles")
```

File: test/battle/move_effect/sp_atk_two_down.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(MoveHasMoveEffectSelf(MOVE_OVERHEAT, MOVE_EFFECT_SP_ATK_TWO_DOWN));
+}
+
+SINGLE_BATTLE_TEST("Single Battle: Overheat drops Sp. Atk by 2 stages")
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/sp_atk_two_down.c:9]:
```suggestion
SINGLE_BATTLE_TEST("Overheat drops Sp. Atk by 2 stages - singles")
```

File: test/battle/move_effect/sp_atk_two_down.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_OVERHEAT, player);
+        HP_BAR(opponent);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
+        MESSAGE("Wobbuffet's Sp. Atk harshly fell!");
+    } THEN {
+        EXPECT_EQ(player->statStages[STAT_SPATK], DEFAULT_STAT_STAGE - 2);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Double Battle: Overheat drops Sp. Atk by 2 stages")
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/sp_atk_two_down.c:26]:
```suggestion
DOUBLE_BATTLE_TEST("Overheat drops Sp. Atk by 2 stages - doubles")
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by AlexOn1ine** [comment]
all done

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**